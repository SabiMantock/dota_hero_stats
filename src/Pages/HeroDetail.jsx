import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import Card from '../components/Card/Card';

const HeroDetail = () => {
  const { heroStats } = useSelector((state) => state.heroStats);
  const { heroName } = useParams();
  const heroStat = heroStats.find((stat) => stat.localized_name === heroName);

  if (!heroStats) {
    return <p>Loading</p>;
  }
  return (
    <div style={{ marginTop: '100px' }}>
      <Card
        name={heroStat.localized_name}
        picks={heroStat.turbo_picks}
        image={heroStat.img}
        roles={heroStat.roles}
        wins={heroStat.turbo_wins}
        type={heroStat.attack_type}
        details={false}
      />
    </div>
  );
};

export default HeroDetail;
