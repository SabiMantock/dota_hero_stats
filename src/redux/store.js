import { configureStore } from '@reduxjs/toolkit';
import heroStatsReducer from './hero_stats';

const store = configureStore({
  reducer: {
    heroStats: heroStatsReducer,
  },
});

export default store;
