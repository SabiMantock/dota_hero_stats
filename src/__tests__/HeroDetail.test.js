import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';

import store from '../redux/store';
import HeroDetail from '../Pages/HeroDetail';

test('renders HeroDetail component', () => {
  const heroTest = render(
    <Provider store={store}>
      <Router>
        <HeroDetail />
      </Router>
    </Provider>,
  );
  expect(heroTest).toMatchSnapshot();
});
