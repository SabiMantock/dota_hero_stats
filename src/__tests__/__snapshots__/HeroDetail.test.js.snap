// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HeroDetail component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="margin-top: 100px;"
      >
        <div
          style="width: 300px; background-color: rgb(206, 63, 117); height: 350px; border-radius: 10px; margin: 50px auto; color: rgb(255, 255, 255);"
        >
          <div>
            <img
              alt="hero_name"
              src="https://api.opendota.comundefined"
              style="width: 100%; border-top-right-radius: 10px; border-top-left-radius: 10px;"
            />
          </div>
          <div
            style="padding: 10px;"
          >
            <h3
              style="color: rgb(255, 255, 255);"
            />
            <div
              style="margin-top: 5px;"
            >
              <span
                style="font-weight: 700;"
              >
                Attack Type: 
              </span>
              <span />
            </div>
            <div
              style="margin-top: 5px;"
            >
              <span
                style="font-weight: 700;"
              >
                Roles: 
              </span>
              <span />
            </div>
            <div
              style="margin-top: 5px;"
            >
              <span
                style="font-weight: 700;"
              >
                Turbo Picks: 
              </span>
              <span />
            </div>
            <div
              style="margin-top: 5px;"
            >
              <span
                style="font-weight: 700;"
              >
                Turbo wins: 
              </span>
              <span />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="margin-top: 100px;"
    >
      <div
        style="width: 300px; background-color: rgb(206, 63, 117); height: 350px; border-radius: 10px; margin: 50px auto; color: rgb(255, 255, 255);"
      >
        <div>
          <img
            alt="hero_name"
            src="https://api.opendota.comundefined"
            style="width: 100%; border-top-right-radius: 10px; border-top-left-radius: 10px;"
          />
        </div>
        <div
          style="padding: 10px;"
        >
          <h3
            style="color: rgb(255, 255, 255);"
          />
          <div
            style="margin-top: 5px;"
          >
            <span
              style="font-weight: 700;"
            >
              Attack Type: 
            </span>
            <span />
          </div>
          <div
            style="margin-top: 5px;"
          >
            <span
              style="font-weight: 700;"
            >
              Roles: 
            </span>
            <span />
          </div>
          <div
            style="margin-top: 5px;"
          >
            <span
              style="font-weight: 700;"
            >
              Turbo Picks: 
            </span>
            <span />
          </div>
          <div
            style="margin-top: 5px;"
          >
            <span
              style="font-weight: 700;"
            >
              Turbo wins: 
            </span>
            <span />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
